### PR TITLE
Add Grafana cloud regions API

### PR DESCRIPTION
**Pull request by julienduchesne**
Structs and functions for this API: https://grafana.com/docs/grafana-cloud/reference/cloud-api/#list-regions

### CODE REVIEW

**Review by matthewnolf — APPROVED** [review]
LGTM 👍

File: cloud_regions.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	HTClusterURL  string `json:"htClusterUrl"`
+}
+
+// CloudRegionsResponse represents the response from the Grafana Cloud regions API.
+type CloudRegionsResponse struct {
+	Items []CloudRegion `json:"items"`
+}
+
+// CloudsRegions fetches and returns all Grafana Cloud regions.
+func (c *Client) CloudsRegions() (CloudRegionsResponse, error) {
```
Comment (matthewnolf) [inline cloud_regions.go]:
```suggestion
func (c *Client) CloudRegions() (CloudRegionsResponse, error) {
```

File: cloud_regions.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	var regions CloudRegionsResponse
+	err := c.request("GET", "/api/stack-regions", nil, nil, &regions)
+	return regions, err
+}
+
+// CloudsRegionBySlug fetches and returns the cloud region which matches the given slug.
+// You can also provide a numeric region ID.
+func (c *Client) CloudsRegionBySlug(slug string) (CloudRegion, error) {
```
Comment (matthewnolf) [inline cloud_regions.go]:
```suggestion
func (c *Client) CloudRegionBySlug(slug string) (CloudRegion, error) {
```

File: cloud_regions.go (hunk 1 above)
Comment (julienduchesne) [reply cloud_regions.go]: oops

**Pull request merged**